src/main/java/com/manardenza/service/HotelService.java

```diff
@@ -26,22 +26,24 @@ public static HotelService getInstance() {
     }
 
     public List<Hotel> findHotelByName(String hotelName) {
-        return new ArrayList<>();
+        return hotelDao.getHotelsByName(hotelName);
     }
 
     public List<Hotel> findHotelByCity(String cityName) {
-        return new ArrayList<>();
+        return hotelDao.getHotelsByCity(cityName);
     }
 
-    public Room bookRoom(long roomId, long hotelId, Reservation reservation) {
-        return null;
+    public Map<String, List<Room>> findPreliminaryRoom(String city, int persons, int price) {
+        Map<String, List<Room>> map = new HashMap<>();
+        hotelDao.getHotelsByCity(city).forEach(hotel -> map.put(hotel.getName(), findRoom(hotel, persons, price)));
+        return map;
     }
 
-    public boolean cancelReservation(long roomId, long userId, long hotelId, Reservation reservation) {
-        return false;
-    }
-
-    public List<Room> findRoom(Map<String, String> params) {
-        return new ArrayList<>();
+    private List<Room> findRoom(Hotel hotel, int persons, int price) {
+        List<Room> test = hotel.getRooms().stream()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "test". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)